controllers/promiserelease_controller_test.go

```diff
@@ -402,4 +405,92 @@ var _ = Describe("PromiseReleaseController", func() {
 			Expect(promise.Spec).To(Equal(expectedPromise.Spec))
 		})
 	})
+
+	// This test is separated from the original nested tests in order to test using mocked kubeclient's method(s)
+	When("the PromiseRelease is installed", func() {
+		var (
+			promiseNamespacedName types.NamespacedName
+			eventRecorder         *record.FakeRecorder
+		)
+
+		BeforeEach(func() {
+			promiseNamespacedName = types.NamespacedName{
+				Name: promise.Name,
+				// Note: Promise is cluster-scoped
+			}
+
+			fakeK8sClient = &mockPromiseCtrlRuntimeClient{
+				Client:  fakeK8sClient,
+				promise: promise.Name,
+				mockErr: errors.NewInvalid( // Create mock error for invalid promise
+					promise.GroupVersionKind().GroupKind(),
+					promise.Name,
+					field.ErrorList{},
+				),
+			}
+
+			eventRecorder = record.NewFakeRecorder(1024)
+
+			reconciler = &controllers.PromiseReleaseReconciler{
+				Client:         fakeK8sClient,
+				Scheme:         scheme.Scheme,
+				PromiseFetcher: fakeFetcher,
+				Log:            ctrl.Log.WithName("controllers").WithName("PromiseRelease"),
+				EventRecorder:  eventRecorder,
+			}
+		})
+
+		When("the sourceRef type is http", func() {
+
+			When("the url fetcher succeeds", func() {
+
+				When("the Promise manifest is invalid", func() {
+
+					BeforeEach(func() {
```

Comment:
Too many blank lines?

Reply:
Saw some tests leaving gaps after `When` and thought they were following it as a convention. But on second thought, I think it is not required. Done. Thanks!